src/Symfony/Component/Messenger/Bridge/Kafka/Transport/KafkaTransportFactory.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Messenger\Bridge\Kafka\Transport;
+
+use Psr\Log\LoggerInterface;
+use Psr\Log\NullLogger;
+use Symfony\Component\Messenger\Exception\InvalidArgumentException;
+use Symfony\Component\Messenger\Transport\Serialization\SerializerInterface;
+use Symfony\Component\Messenger\Transport\TransportFactoryInterface;
+use Symfony\Component\Messenger\Transport\TransportInterface;
+
+/**
+ * @author Konstantin Scheumann <konstantin@konstantin.codes>
+ */
+class KafkaTransportFactory implements TransportFactoryInterface
+{
+    private const DSN_PROTOCOLS = [
+        self::DSN_PROTOCOL_KAFKA,
+        self::DSN_PROTOCOL_KAFKA_SSL,
+    ];
+    private const DSN_PROTOCOL_KAFKA = 'kafka://';
+    private const DSN_PROTOCOL_KAFKA_SSL = 'kafka+ssl://';
```

Comment:
when is the ssl mode enforced when this scheme is used?
can't we reduce the number of constants? It looks like a lot of indirection to me, with extra boilerplate (the foreach). Please check how the other transports do and follow their style.